command/agent/agent.go

```diff
@@ -935,7 +935,9 @@ func (a *Agent) setupClient() error {
 // If no HTTP health check can be supported nil is returned.
 func (a *Agent) agentHTTPCheck(server bool) *structs.ServiceCheck {
 	// Resolve the http check address
-	httpCheckAddr := a.config.normalizedAddrs.HTTP
+	// TODO: evaluate if this is the best way to handle this default, maybe
+	// BindAddr?
+	httpCheckAddr := a.config.normalizedAddrs.HTTP[0]
```

Comment:
The [`checks_use_advertise`](https://www.nomadproject.io/docs/configuration/consul#checks_use_advertise) docs imply that using `normalizedAddrs.HTTP` is a bug, but given that `addresses.http` is supposed to be overriding `bind_addr`, I think we should keep the behavior as you have it here and fix the docs for `checks_use_advertise` to describe the fallback from `addresses.http` to `bind_addr`.

Reply:
Took an initial stab at updating the documentation! let me know what you think.

Reply:
Docs look good.